Host utilities: bail early when dns_get_record is not defined.

jeherve opened the pull request:
## Proposed changes:

This should avoid errors in environments where `dns_get_record()` is not available, like on WordPress Playground at the moment:
https://github.com/WordPress/wordpress-playground/issues/1042

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See p1HpG7-rqu-p2

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* I do not quite know how to test this. Once it is merged, we will be able to test it with WordPress Playground.

jeherve committed:
Host utilities: bail early when dns_get_record is not defined.

See p1HpG7-rqu-p2

darssen commented on projects/packages/status/src/class-host.php:
```diff
@@ -142,9 +142,13 @@ public function get_source_query() {
 	 * @return string
```
Since we're at it, should we change `string` here to `array`?

jeherve replied:
Good point. Done in 17671f27502dc16b0223e92e029f6b95e27a090c

darssen commented on projects/packages/status/src/class-host.php:
```diff
@@ -142,9 +142,13 @@ public function get_source_query() {
 		$nameservers = array();
 
+		if ( ! function_exists( 'dns_get_record' ) ) {
+			return $nameservers;
```
If we are returning early, I think it makes more sense to be explicit and return an empty array directly.
Then maybe move the `$nameservers` declaration after the check for clarity purposes to.

jeherve replied:
Done in 17671f27502dc16b0223e92e029f6b95e27a090c

darssen reviewed (commented):
Thanks for taking care of this @jeherve!
I made a couple of minor comments, which shouldn't be blockers though.

jeherve committed:
Address review feedback

- https://github.com/Automattic/jetpack/pull/36019#discussion_r1505734224
- https://github.com/Automattic/jetpack/pull/36019#discussion_r1505736311
Co-authored-by: Juanma Rodriguez Escriche <juanma.rodriguez.escriche@automattic.com>

darssen reviewed (approved):
LGTM!
